<?xml version="1.0"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

<property name="M_PI" value="3.1415926535897931" />

  <xacro:macro name="wam6" params="parent">

    <link name="lk_wam6">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0" />
             <mass value="0.4278" />
            <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="0.1"  iyz="0.0"  izz="0.1" />
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
            <geometry name="lk_wam6_visual">
                <mesh filename="package://iri_wam_description/mesh/wam6.stl" scale="1.0 1.0 1.0" />
            </geometry>
            <material name="Grey" >
                <color rgba="0.75 0.75 0.75 1.0"/>
            </material>
        </visual>
        <collision>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry name="lk_wam6_collision">
                <mesh filename="package://iri_wam_description/mesh/wam6.stl" scale="1.0 1.0 1.0" />
<!--                <box size="0.10 0.10 0.10"/> -->
            </geometry>
            <contact_coefficients kd="1.0" kp="1000.0" mu="0"/>
        </collision>
    </link>
    <gazebo reference="lk_wam6">
        <material>Gazebo/Grey</material>
        <selfCollide>true</selfCollide>
    </gazebo>

    <joint name="j6_joint" type="revolute">
        <parent link="${parent}"/>
        <child link="lk_wam6"/>
        <origin xyz="0.0 0.0 0.0" rpy="-${M_PI/2} 0 0" />
        <axis xyz="0 0 1" />
        <limit lower="-1.6" upper="1.6" effort="30" velocity="2.0"/>
        <!--  <safety_controller k_velocity="0.5"/>-->
        <dynamics damping="100"/>
    </joint>
    
    <transmission type="pr2_mechanism_model/SimpleTransmission" name="j6_transmission">
      <actuator name="j6" />
      <joint name="j6_joint" />
      <mechanicalReduction>1</mechanicalReduction>
      <motorTorqueConstant>1</motorTorqueConstant>
    <!--  <pulsesPerRevolution>90000</pulsesPerRevolution>-->
    </transmission>

  </xacro:macro>

</robot>
